### PR TITLE
feat: copy code

### PR DESCRIPTION
**Pull request by zaaakher**
Related to #238

### CODE REVIEW

**Comment by zaaakher** [comment]
I think we still need a tooltip/popover as a feedback that the code has been copied.

File: src/components/markdown.tsx [hunk 1]
```diff
@@ -572,6 +575,8 @@ function Image(props: React.ComponentPropsWithoutRef<"img">) {
 }
 
 function Code({ className, inline, children }: CodeProps) {
+  const { copy } = useClipboard()
```
Comment (colebemis) [inline src/components/markdown.tsx:578]: We shouldn't need this `useClipboard()` hook. We're already using a `copy-to-clipboard` package in other parts of the code base:

https://github.com/lumen-notes/lumen/blob/main/src/components/note-card.tsx#L3

File: src/components/markdown.tsx [hunk 2]
```diff
@@ -588,9 +593,18 @@ function Code({ className, inline, children }: CodeProps) {
...
 
   return (
-    <pre>
-      <SyntaxHighlighter language={language}>{children}</SyntaxHighlighter>
-    </pre>
+    <div className="relative">
+      <pre>
+        <IconButton
+          onClick={() => copy(children)}
+          aria-label="copy code"
```
Comment (colebemis) [inline src/components/markdown.tsx:600]:
```suggestion
          aria-label="Copy"
```

**Review by colebemis — COMMENTED** [review]
Thanks for the PR, @zaaakher! 🙌 

Do you have any ideas about how to handle long text that appears under the copy button?

CleanShot 2024-01-09 at 21 04 24@2x

**Comment by zaaakher** [comment]
> Do you have any ideas about how to handle long text that appears under the copy button?

Perhaps we can add `pe-10` to the content of the `<pre>`  to add padding at the end so that the copy icon is not covering the text?

**Comment by zaaakher** [comment]
> Perhaps we can add pe-10 to the content of the <pre> to add padding at the end so that the copy icon is not covering the text?

Actually no that wouldn't look good if there are more than 1 line in the codeblock. This problem only happens to the first line of the content of the codeblock. If we add the end-padding it will look good when it's only 1 line but won't look good when more than 1 line.

**Comment by zaaakher** [comment]
image
Actually it doesn't look bad and works well.
Although I had to do `style={{ paddingInlineEnd: 50 }}` because tailwind's `pe-12` classname didn't work for some reason

**Review by colebemis — APPROVED** [review]
Looks great. Thank you!

**Comment by zaaakher** [comment]
You're welcome 👍 

@colebemis if I ever get around making a popover that says "copied", should I place it inside the `IconButton` next to the `shortcut` condition? or make it stand-alone and wrap the `IconButton` in this PR with the new popover?

**Comment by colebemis** [comment]
@zaaakher I don't think it needs to be a popover. I like the idea of having it switch to a checkmark icon temporarily instead (which also seems simpler to implement).



https://github.com/lumen-notes/lumen/assets/4608155/49ed5908-94a0-4fe7-acd8-055d725438d9

**Comment by zaaakher** [comment]
Agree, much simpler 👍 just implemented it in #241

**Pull request merged**